index/scorch/merge.go

```diff
@@ -159,6 +159,20 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 	// process tasks in serial for now
 	var notifications []chan *IndexSnapshot
 	var filenames []string
+	// clean up any pending notifications from introducer on exit
+	// from an index closure.
+	defer func() {
+		for _, notification := range notifications {
+			if len(notification) > 0 {
+				newSnapshot := <-notification
+				atomic.AddUint64(&s.stats.TotFileMergeIntroductionsDone, 1)
+				if newSnapshot != nil {
+					_ = newSnapshot.DecRef()
+				}
+			}
+		}
+	}()
+
```

Comment:
also, another thought... now that you have a defer here that consumes the notifications... wondering if the other loop in the main codepath that consumes the notifications might be removable or consolidate'able?

Reply:
Not sure of removing it as we need the blocking behaviour there for the merge introductions, I admit that I am very poor in visualising any potential consolidations here.. :)